### PR TITLE
b-enhancment/gh-1248-manual-adjust-allows-large-negative-numbers

### PR DESCRIPTION
**Pull request by popovicbojan**
Resolves: #1248

### CODE REVIEW

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java [hunk 1]
```diff
@@ -20,15 +23,19 @@
   private String comment;
 
   @ValidAllowance(message = "'Manual adjust' imputed value must be divisible by 4.")
+  @Range(min = -240, max = 240, message = "Manual adjust value must be between -30 and 30. ")
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Message is very technic. Please update the message to `Manual adjustment can be done for maximum 30 days.`.

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java [hunk 2]
```diff
@@ -20,15 +23,19 @@
...
   private int manualAdjust;
 
   @ValidAllowance(message = "'Bonus manual adjust' imputed value must be divisible by 4.")
+  @Range(min = 0, max = 40, message = "Bonus days (manual) value must be between 0 and 5. ")
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Please update the message to be more user friendly - `You can ask for a maximum of 5 bonus days.`.

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java [hunk 3]
```diff
@@ -20,15 +23,19 @@
...
   private int bonusManualAdjust;
 
   @ValidAllowance(message = "'Carried over' imputed value must be divisible by 4.")
+  @Range(min = 0, max = 40, message = "Carried over value must be between 0 and 5. ")
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Please update the message to be more user friendly.

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java [hunk 4]
```diff
@@ -20,15 +23,19 @@
...
   private int carriedOver;
 
   @ValidAllowance(message = "'Training manual adjust' imputed value must be divisible by 4.")
+  @Range(min = -16, max = 16, message = "Training manual adjust value must be between -2 and 2. ")
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Please update the message to be more user friendly.

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java (hunk 2 above)
Comment (popovicbojan) [reply src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: All these exception messages are related to user allowance manual adjustment which are edited by administration, users cannot edit it for themselves. So I propose message:
`Employee can be granted maximum 5 bonus days.`

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java (hunk 3 above)
Comment (popovicbojan) [reply src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Is it ok to write this message:
`Employee can carry over a maximum of 5 days.`

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java (hunk 4 above)
Comment (popovicbojan) [reply src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Proposition:
`Manual adjustment of training leave can be done for a maximum of 2 days.`

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java (hunk 2 above)
Comment (stanislavgrujic) [reply src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Administration is also a user. :) I agree with the proposed message.

Comment (popovicbojan) [reply src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: I agree. I meant user with role USER. :)

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java [hunk 5]
```diff
@@ -20,15 +21,19 @@
   private String comment;
 
   @ValidAllowance(message = "'Manual adjust' imputed value must be divisible by 4.")
+  @Range(min = -240, max = 240, message = "Manual adjustment can be done for maximum 30 days. ")
   private int manualAdjust;
 
   @ValidAllowance(message = "'Bonus manual adjust' imputed value must be divisible by 4.")
+  @Range(min = 0, max = 40, message = "Employee can be granted maximum 5 bonus days. ")
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Please update the message to: *Employee can be granted for a maximum of 5 bonus days.*

File: src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java [hunk 6]
```diff
@@ -20,15 +21,19 @@
   private String comment;
 
   @ValidAllowance(message = "'Manual adjust' imputed value must be divisible by 4.")
+  @Range(min = -240, max = 240, message = "Manual adjustment can be done for maximum 30 days. ")
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/dto/AllowanceEditDto.java]: Please update the message to *Manual adjustment can be done for a maximum of 30 days.*

**Pull request merged**